patches/net/minecraft/server/network/config/SynchronizeRegistriesTask.java.patch

```diff
@@ -0,0 +1,17 @@
+--- a/net/minecraft/server/network/config/SynchronizeRegistriesTask.java
++++ b/net/minecraft/server/network/config/SynchronizeRegistriesTask.java
+@@ -45,11 +_,9 @@
+     }
+ 
+     public void handleResponse(List<KnownPack> p_326076_, Consumer<Packet<?>> p_326167_) {
+-        if (p_326076_.equals(this.requestedPacks)) {
+-            this.sendRegistries(p_326167_, Set.copyOf(this.requestedPacks));
+-        } else {
+-            this.sendRegistries(p_326167_, Set.of());
+-        }
++        Set<KnownPack> requested = new java.util.HashSet<>(this.requestedPacks);
++        requested.retainAll(p_326076_);
++        this.sendRegistries(p_326167_, requested);
```

Comment:
What motivated this change? (Should we add a comment here?)

Reply:
Good point, I'll add a comment. Basically, beforehand, you needed an _exact_ match of all requested known packs being present on the client for _any_ to be used. With this, it'll use the ones that are present and sync the ones that are absent the normal way - which is necessary in the case of, say, datapacks-as-mods present only on the server, to avoid breaking KnownPack syncing entirely in that case